pkg/config/structure/unmarshal_test.go

```diff
@@ -1578,3 +1579,24 @@ network_devices:
 	assert.NoError(t, err)
 	assert.Equal(t, []string{"a", "b", "c"}, actualStrings)
 }
+
+func TestUnmarshalKeyOnSliceOfMap(t *testing.T) {
+	t.Setenv("DD_TEST_VALUE", "[{\"a\": \"1\", \"b\": \"1\"}, {\"a\": \"2\", \"b\": \"2\"}]")
+	mockConfig := newEmptyMockConf(t)
+
+	mockConfig.BindEnvAndSetDefault("test_value", []map[string]int{}) //nolint:forbidigo // TODO: replace by 'SetDefaultAndBindEnv'
```

Comment:
This TODO in the linter comment uses a function name that doesn't exist. But also I don't think we need the `nolint` at all since this method isn't deprecated.